ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -458,6 +479,52 @@ def yield_view_lineage(self) -> Optional[Iterable[AddLineageRequest]]:
                     f"Could not parse query [{view_definition}] ingesting lineage failed: {exc}"
                 )
 
+    def _get_foreign_constraints(
+        self, table_constraints: OMetaTableConstraints
+    ) -> List[TableConstraint]:
+        """
+        Search the referred table for foreign constraints
+        and get referred column fqn
+        """
+
+        foreign_constraints = []
+        for constraint in table_constraints.foreign_constraints:
+            referred_column_fqns = []
+            referred_table = fqn.search_table_from_es(
+                metadata=self.metadata,
+                table_name=constraint.get("referred_table"),
+                schema_name=constraint.get("referred_schema"),
+                database_name=None,
+                service_name=self.context.database_service.name.__root__,
+            )
```

Comment:
I am wondering if we could cache that info (maybe a dict outside the loop that we'll delete once we have finished  processing our constraints). I think that es calls are expensive and we could potentially, if I well understand, have multiple constraints refering to the same table. WDYT?

Reply:
Umm that is true and now since you have pointed this now I can think of a way where I will not need to make any es call let me see if that works!